extract/bbox.py

```diff
@@ -46,12 +27,14 @@ class BBoxMixin(object):
 
     @property
     def bbox(self):
+        if 'bbox' not in self.attr:
+            return
```

Comment:
This feels wrong to me. It means the bbox function sometimes returns a BBox but sometimes returns nothing, and the caller is going to have to check what they got back.

```python
is_it_a_bb = thing.bbox()
if is_it_a_bb == nil:
    thing
else:
    other thing
````

That's maybe not terrible, because you can probably compact it to:

````python
bb = thing.bbox()
if bb:
````

because the nothing that gets returned here should be falsy, I think. But it still makes it easy to later call `bb.method()` and get `NoneType has no such method` errors.

I feel like maybe it would be better to throw an exception here (see https://docs.python.org/3.7/tutorial/errors.html#user-defined-exceptions), if not having a bbox is going to be rare enough (and what to do in this case undefined enough) that it ought to require special handling to recover.

But if this is actually going to be common, then an explicit `return None` is probably fine.

Reply:
But, similar to my comments on use of `continue` - I'd prefer to have the main code body in the `true` branch.
````python
`if `bbox` in self.attr:
    *lots of code*
````

Note that doing it this way means you don't need to explicitly 'return None'

````python
>>> def func(thing):
...     if thing:
...         return thing
...     
... 
>>> print(func("yes"))
yes
>>> print(func(""))
None
>>> 
````